src/EnergyPlus/OutputProcessor.cc

```diff
@@ -3396,7 +3396,8 @@ void UpdateDataandReport(EnergyPlusData &state, OutputProcessor::TimeStepType co
         }
 
         // Update meters on the TimeStep  (Zone)
-        if (op->meterValues.capacity() > 0) {
+        if (op->meterValues.capacity() > 0 and
+            (!state.dataSysVars->ReportDuringWarmup || (state.dataSysVars->ReportDuringWarmup and !state.dataGlobal->WarmupFlag))) {
```

Comment:
When did we change to using `and` instead of `&&`?  Instead of holding off meter reporting I think the meters need to be reset after the warmup period (i.e., you shouldn't need to change this line). That must be done somewhere?

                // for simulations that last longer than a week, identify when the last year of the simulation is started
                if ((state.dataGlobal->DayOfSim > 365) && ((state.dataGlobal->NumOfDayInEnvrn - state.dataGlobal->DayOfSim) == 364) &&
                    !state.dataGlobal->WarmupFlag) {
                    DisplayString(state, "Starting last  year of environment at:  " + state.dataGlobal->DayOfSimChr);
                    OutputReportTabular::ResetTabularReports(state);
                }

Reply:
I guess that this can be done with `ResetAccumulationWhenWarmupComplete`. A cursory search shows that it is not currently being used.

Reply:
I just pushed some changes to do the reset after the warmup period instead. Both this approach and the approach from the initial commit "fix" the issue: all non-HVAC end-use energy consumption is the same whether or not `ReportDuringWarmup` is used, but some of the HVAC end-use energy consumption is still slightly different. I'm not sure **yet** what the cause of it is. I tried a simpler model (without a plant loop) and don't see the issue, (i.e., with this fix the report energy use is identical whether or not `ReportDuringWarmup` is used). Thoughts?

Reply:
As far as I can tell they were not being reset after the warmup period, I could see large values associated with the meters right at the beginning of the simulation.

I pushed some changes to reset them after the warmup period and I get the same results as with the initial changes (i.e., much better than with develop, but still diffs for HVAC end-uses).

What's interesting is that the first few days of the simulation show identical results, they only start to deviate after about 10 days for the defect file. When I say that the results are different it's not just the reported meter values, but actual equipment level variables (fans, boiler, etc.). I'm not yet sure how resetting meters (or holding off from reporting) would lead to equipment behaving differently. I'll keep looking.